src/main/java/com/lorepo/icplayer/client/xml/content/parsers/ContentParserBase.java

```diff
@@ -134,8 +134,10 @@ protected String parsePageName(Element rootElement, String pageNode) {
 		for(int i = 0; i < descriptorNodes.getLength(); i++){
 			Element node = (Element)descriptorNodes.item(i);
 			String addonId = node.getAttribute("addonId");
-			String href = StringUtils.unescapeXML(node.getAttribute("href"));
-			addonDescriptors.put(addonId, new AddonDescriptor(addonId, href));
+			if (addonId != "null") {
```

Comment:
Po pierwsze, ta sama uwaga co wyżej odnośnie equals. Po drugie, na przyszłość sugeruję być bardzo ostrożnym przy edytowaniu parserów - generalnie jeżeli wprowadzana zmiana zmieniłaby format xml'a albo w inny sposób zakłócała kompatybilność wsteczną, to trzeba utworzyć nową wersję (np. ContentParser_v3.java, analogicznie do poprzednich wersji). W tym wypadku raczej chyba nie jest to potrzebne, bo to tylko fix/zabezpieczenie przed sytuacją, która nie powinna w ogóle występować.